src/id3/v2/tag.rs

```diff
@@ -547,21 +607,54 @@ impl SplitAndMergeTag for ID3v2Tag {
 		fn split_pair(
 			content: &str,
 			tag: &mut Tag,
-			current_key: ItemKey,
+			number_key: ItemKey,
 			total_key: ItemKey,
 		) -> Option<()> {
+			fn parse_number(source: &str) -> Option<&str> {
+				let number = source.trim();
+
+				if number.is_empty() {
+					return None;
+				}
+
+				if str::parse::<u32>(number).is_ok() {
+					Some(number)
+				} else {
+					log::warn!("{number:?} could not be parsed as a number.");
+
+					None
+				}
+			}
+
 			let mut split =
 				content.splitn(2, &[V4_MULTI_VALUE_SEPARATOR, NUMBER_PAIR_SEPARATOR][..]);
-			let current = split.next()?.to_string();
-			tag.items
-				.push(TagItem::new(current_key, ItemValue::Text(current)));
 
-			if let Some(total) = split.next() {
-				tag.items
-					.push(TagItem::new(total_key, ItemValue::Text(total.to_string())))
-			}
+			let number_source = split.next()?;
+			let number = parse_number(number_source)?;
+			let total_source = split.next();
+			let total = total_source.and_then(parse_number);
+
+			let is_valid = if let Some(total_source) = total_source {
```

Comment:
This check doesn't work. You will notice if your add tests for leading/trailing whitespace.

Implicitly assuming that the length of NUMBER_PAIR_SEPARATOR equals 1 byte is brittle.

Reply:
Please allow me to confirm the details.

> You will notice if you add tests for leading/trailing whitespace.

What test cases do you think? The followings are passed as a valid content when I added the unit test:

* " 1 "
* "&nbsp;&nbsp;1&nbsp;&nbsp;" (The spaces have 2 characters)
* "1&nbsp;&nbsp;/&nbsp;&nbsp;2" (The spaces have 2 characters)
* "&nbsp;&nbsp;1&nbsp;&nbsp;/&nbsp;&nbsp;2&nbsp;&nbsp;" (The spaces have 2 characters)

Because `content` is checked the following:

* A not trimmed string has a number only.
* One separator character and two not trimmed strings that have a number only.

> Implicitly assuming that the length of NUMBER_PAIR_SEPARATOR equals 1 byte is brittle.

May do not NUMBER_PAIR_SEPARATOR equals 1 bytes? I would be glad if you could provide an example.

Otherwise, do you mean that the constant should be extracted?

Reply:
The magic number number 1 appears out of the context of `NUMBER_PAIR_SEPARATOR`. If the constant was of type `str` you could use `NUMBER_PAIR_SEPARATOR.len()`. But using the size of the separator is probably no longer needed after fixing the parsing. It should not be relevant for the decision, only number and total with their parse results need to be considered.

Reply:
Returning an `Option` from parse_number() is probably no sufficient, because you need to distinguish between _empty_ and _not a number_.

Reply:
Using the low-level input representation again for making the final decision is conceptually unsound, different levels of abstraction.

Reply:
Indeed, I overlooked some tests. But I am still unsure that this code covers all cases correctly, especially the expression `content.len() == number_source.len() + 1 + total_source.len()`. I already mentioned the various reasons why it is not suitable here.

Reply:
Thank you for details. I understand that I should improve parsing about `content`. I will consider it.